Marlin/src/module/tool_change.cpp

```diff
@@ -956,13 +956,13 @@ void tool_change(const uint8_t tmp_extruder, bool no_move/*=false*/) {
 
         #if ENABLED(SINGLENOZZLE)
           #if FAN_COUNT > 0
-            singlenozzle_fan_speed[active_extruder] = thermalManager.fan_speed[0];
-            thermalManager.fan_speed[0] = singlenozzle_fan_speed[tmp_extruder];
+            singlenozzle_fan_speed[!active_extruder] = thermalManager.fan_speed[0];
+            thermalManager.fan_speed[0] = singlenozzle_fan_speed[active_extruder];
           #endif
 
-          singlenozzle_temp[active_extruder] = thermalManager.temp_hotend[0].target;
-          if (singlenozzle_temp[tmp_extruder] && singlenozzle_temp[tmp_extruder] != singlenozzle_temp[active_extruder]) {
-            thermalManager.setTargetHotend(singlenozzle_temp[tmp_extruder], 0);
+          singlenozzle_temp[!active_extruder] = thermalManager.temp_hotend[0].target;
```

Comment:
are you sure of this "!" ? uint16_t singlenozzle_temp[EXTRUDERS];
its always 2 ?

Reply:
I followed how thinky corrected some of this logic in the past. Personally, id like to set an old_tool variable on entry to track absolute of new and old in all situations. At least with 2 extruders this works, but 3 or 5 to a singlenozzle i know it'll break.

Reply:
@tpruvot just tossed up the way I had originally done it when I first added the feature, so it doesnt matter if active tool was already changed. Ill leave it to thinky to use which of the two he wants. Also pulled out a dead line, since it was already moved above...